crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -113,16 +117,22 @@ pub trait L2BlockSource: Send + Sync {
     async fn get_l2_block(&mut self, height: BlockHeight) -> Result<BlockInfo>;
 }
 
-#[derive(Debug, Default)]
-pub struct DaCommitDetails {
+#[derive(Debug, Default, Clone)]
+pub struct DaGasPrice {
     pub l2_block_range: core::ops::Range<u32>,
     pub blob_size_bytes: u32,
     pub blob_cost_wei: u32,
-    pub partial_block_heights: Option<[u32; 2]>,
 }
 
-pub trait DaCommitSource: Send + Sync {
-    fn get_da_commit_details(&mut self) -> Result<Option<DaCommitDetails>>;
+pub trait GetDaGasPriceFromSink: Send + Sync + Default + Clone {
+    fn get(&mut self) -> Result<Option<DaGasPrice>>;
+}
+
+// ensures that the da gas price is set to the sink only by
+// selected impls. we don't want the algo updater to have
+// access to this method.
+pub trait SetDaGasPriceToSink: Send + Sync + Default + Clone {
```

Comment:
I don't think this trait needs to exist. 

(Committer <- CommitterDaBlockCostSource Adapter -> (DaBlockCostSource Port <- DaBlockCostService -> `Arc<Mutex<_>>` <- DaGasPriceProvider Adapter -> (GetDaBlockCost Port <- GasPriceUpdaterService)))

Arrows are in the direction of dependencies. **The `DaBlockCostService` is part of the `DaGasPriceProvider` adapter.** We don't need to isolate `DaBlockCostService` from `Arc<Mutex<_>>` since it's job is to get data from some `DaBlockCostSource` and write to an `Arc<Mutex<_>>`. We insulate ourselves from the dependency on `Arc<Mutex<_>>` on the get side.

I've added the "((()))" around the design to show why Ports and Adapters is often called "onion architecture": https://medium.com/expedia-group-tech/onion-architecture-deed8a554423 
It's because the dependencies go outward, not inward, from the domain. Also related:
https://en.wikipedia.org/wiki/Dependency_inversion_principle

Reply:
addressed in https://github.com/FuelLabs/fuel-core/pull/2163/commits/74fc9bf4e08cb0d4a1a5def23f31d2c4fbd31c9f#diff-d0929c84f197be22ea7a8f1bc32981a49731a6ec12250819e9c86ebba1373ffaL43-L54